extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -424,6 +430,51 @@ export class AmpAnalytics extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Returns a promise that resolves when configuration is re-written if
+   * configRewriter is configured by a vendor.
+   *
+   * @private
+   * @return {!Promise<undefined>}
+   */
+  rewriteConfig_() {
+    const remoteConfigUrl = this.getConfigRewriter_()['url'];
+
+    if (!remoteConfigUrl || this.isSandbox_) {
+      // use default configuration merge.
+      return Promise.resolve();
+    }
+
+    assertHttpsUrl(remoteConfigUrl, this.element);
+    const TAG = this.getName_();
+    dev().fine(TAG, 'Rewriting config', remoteConfigUrl);
+
+    const config = dict({});
+    this.mergeObjects_(this.getInlineConfigNoInline(), config);
+    this.mergeObjects_(this.remoteConfig_, config);
```

Comment:
can we re-organize the code to avoid unnecessary object merges in `mergeConfigs_()`. there is duplicated work.

Reply:
I'm not sure I follow this one.. 
If rewrite config is enabled we merge 
* inline+remote here and 
* vendor default + rewriter defaults + rewritten in mergeConfig_()

otherwise only default configuration merge happens 
vendor default + vendor type + inline + remote.

Reply:
pseudo code:
```
mergedConfig = vendorDefault + inlined + remote
if (rewriter) {
  mergedConfig = rewrite(mergedConfig, rewriterDefaults)
}
return mergedConfig
```